lib/Doctrine/ODM/MongoDB/Hydrator/HydratorFactory.php

```diff
@@ -185,10 +185,13 @@ private function generateHydratorClass(ClassMetadata $class, string $hydratorCla
                     <<<'EOF'
 
         // Field(type: "date")
-        if (isset($data['%1$s'])) {
+        if (isset($data['%1$s']) || (! empty($this->class->fieldMappings['%2$s']['nullable']) && array_key_exists('%1$s', $data))) {
```

Comment:
We avoid usage of `empty` as it covers too many cases. Maybe we can refactor the condition like this;

```suggestion
        if (array_key_exists('%1$s', $data) && ($data['%1$s'] !== null || $this->class->fieldMappings['%2$s']['nullable'] ?? false)) {
```

Reply:
Yes, I understand, I had just copied another part of the factory. In an external project I try to maintain consistency.
I forgeted the flag WIP. I just wanted to run the tests, not being able to do it on my machine

Reply:
Indeed, I wasn't aware that empty is used a lot in this class.